### ISSUE TITLE
Suggest what features you want here

### ISSUE THREAD

**Issue by reisxd** [reisxd/revanced-builder#283]
Hi everyone, I want to implement new and awesome things to Builder so, I ask you, what do you want me to implement to Builder?

Also, soon there'll be an native Android app to build ReVanced even more easily (if i don't fail) ;)

**Comment by Canny1913** [comment]
multi language support

**Comment by reisxd** [comment]
> multi language support

https://user-images.githubusercontent.com/29177546/185799475-df206561-aca3-4378-b6b5-9d7fb4ca7149.mov

**Comment by Canny1913** [comment]
why not

**Comment by reisxd** [comment]
> why not

No one probably willing to contribute and this'll *probably* bloat the frontend.

**Comment by alexandreteles** [comment]
> > why not
> 
> No one probably willing to contribute and this'll _probably_ bloat the frontend.

i18n?

Include/download the appropriate JRE instead of asking the user to install it.

**Comment by afnzmn** [comment]
compose rewrite

**Comment by shuniichan** [comment]
Can you implement function to choose our own name branding while building? Thanks.

**Comment by reisxd** [comment]
> Can you implement function to choose our own name branding while building? Thanks.

Currently impossible to do (it *is* possible but it's complicated).

**Comment by TgZ39** [comment]
View downloaded videos in revanced app (for example in the 
media library) and when clicking on them it switches to powertube app or vlc(?) and automatically starts playing it.

**Comment by reisxd** [comment]
> View downloaded videos in revanced app (for example in the 
> media library) and when clicking on them it switches to powertube app or vlc(?) and automatically starts playing it.

Not something related to Builder.

**Comment by TgZ39** [comment]
> > View downloaded videos in revanced app (for example in the
> > media library) and when clicking on them it switches to powertube app or vlc(?) and automatically starts playing it.
> 
> Not something related to Builder.
then fix it :)))))

**Comment by TgZ39** [comment]
A Discord Server!

**Comment by shrihanDev** [comment]
> Can you implement function to choose our own name branding while building? Thanks.

Wait for me.

**Comment by laltroweb** [comment]
Allow to compile and save root release style nonroot release (atm instead, with android script, is necessary to install before YouTube and when compiled we do not have apk to share).

**Comment by ktheticdev** [comment]
updating root revanced without reinstalling

**Comment by dadazord** [comment]
I would like to see the ability to choose older versions of Revanced Patcher, CLI and ReVanced Integrations, and not just new ones.  New versions are not always stable.

**Comment by HikaruIchijyo** [comment]
> Hi everyone, I want to implement new and awesome things to Builder so, I ask you, what do you want me to implement to Builder?
> 
> Also, soon there'll be an native Android app to build ReVanced even more easily (if i don't fail) ;)

idk what do u mean by native android app but app like aliucord installer or something like that would be awesome instead of manually using termux it just download the latest build decompile & patch it recompile & u r good to go

**Comment by F-AlFaris** [comment]
How can I check which version of revanced-builder I'm running ?

**Comment by freMea** [comment]
1. On PC, check if a folder at the same level of the app is named JDK and use it.
2. If a file named **settings.json** is found, read it to get the patches to apply

**Comment by ReylordFX** [comment]
Custom Themes?

**Comment by tribela** [comment]
changing package name?

**Comment by reisxd** [comment]
> changing package name?

Impossible to do.

**Comment by magmabits** [comment]
i want forg easter egg, thx

**Comment by reisxd** [comment]
> i want forg easter egg, thx
> 

no

~~but therell be an easter egg~~

**Comment by magmabits** [comment]
> > i want forg easter egg, thx
> 
> no
> 
> ~but therell be an easter egg~

mean

**Comment by m-usman-24** [comment]
Can we get the orignal app icon and orignal dark theme as an option like in vanced manager

**Comment by Poggyy** [comment]
Auto Updater

**Comment by squidds** [comment]
Is casting a root only feature? Unfortunately my phone is from a provider that makes it impossible to unlock the bootloader.

**Comment by TgZ39** [comment]
> Auto Updater

would be a very nice feature

**Comment by EatVanced** [comment]
Support feature for patching Vanced
![image](https://user-images.githubusercontent.com/100286130/187032198-d8c086e3-e3d5-44b7-8d71-045ab84d19ed.png)

**Comment by ElissonRodrigues** [comment]
> > por que não
> 
> Ninguém provavelmente está disposto a contribuir e isso _provavelmente_ vai inchar o frontend.

I can contribute with the translation to Portuguese-Brazil

**Comment by WatermelonOof** [comment]
`android-irface.sh` allow you to delete cache (deleting the whole revanced/), but this deletes the keystore too. I hope to see an option to preserve the keystore while removing the real "cache".
P.S. I'm talking about before running the revanced-builder instead during reinstallation.

**Comment by driftywinds** [comment]
Anyway to specify the Icon of the patched app and the name during the building process? It would be really helpful

**Comment by reisxd** [comment]
> Anyway to specify the Icon of the patched app and the name during the building process? It would be really helpful

https://github.com/reisxd/revanced-builder/issues/283#issuecomment-1225872033

**Comment by driftywinds** [comment]
> > Anyway to specify the Icon of the patched app and the name during the building process? It would be really helpful
> 
> [#283 (comment)](https://github.com/reisxd/revanced-builder/issues/283#issuecomment-1225872033)

not even the icon?

**Comment by reisxd** [comment]
> > > Anyway to specify the Icon of the patched app and the name during the building process? It would be really helpful
> > 
> > 
> > [#283 (comment)](https://github.com/reisxd/revanced-builder/issues/283#issuecomment-1225872033)
> 
> not even the icon?

Yes, because of limitations.

**Comment by rafagale** [comment]
root support again please

**Comment by reisxd** [comment]
> root support again please

There is...?

**Comment by rafagale** [comment]
> > root support again please
> 
> There is...?

its not working anymore for me after [v3.3.6 release](https://github.com/reisxd/revanced-builder/releases/tag/v3.3.6). It's all good with 3.3.5 though

**Comment by reisxd** [comment]
> > > root support again please
> > 
> > There is...?
> 
> its not working anymore for me after [v3.3.6 release](https://github.com/reisxd/revanced-builder/releases/tag/v3.3.6). It's all good with 3.3.5 though

What do you mean "not working"? Please create an issue.

**Comment by demiGod095** [comment]
give user option to select mount or non-mount install ?

**Comment by zeGolem** [comment]
Just stumbled on this project… Would be cool to have build instructions for advanced users who don't want to rely on the prebuilt binaries, or want to test features before a new release

**Comment by reisxd** [comment]
> Just stumbled on this project… Would be cool to have build instructions for advanced users who don't want to rely on the prebuilt binaries, or want to test features before a new release

https://github.com/revanced/revanced-documentation is useful for developers ;)

**Comment by zeGolem** [comment]
> > Just stumbled on this project… Would be cool to have build instructions for advanced users who don't want to rely on the prebuilt binaries, or want to test features before a new release
> 
> https://github.com/revanced/revanced-documentation is useful for developers ;)

It indeed is, and I followed these instructions to build revanced myself. But now I want to build *your* project myself. Could you provide instructions in the README how to compile the code in the repo into an executable?

**Comment by reisxd** [comment]
> > > Just stumbled on this project… Would be cool to have build instructions for advanced users who don't want to rely on the prebuilt binaries, or want to test features before a new release
> > 
> > https://github.com/revanced/revanced-documentation is useful for developers ;)
> 
> It indeed is, and I followed these instructions to build revanced myself. But now I want to build *your* project myself. Could you provide instructions in the README how to compile the code in the repo into an executable?

Oh, you meant Builder...😅

Sure, I'll add it!

**Comment by git-Hmmm** [comment]
> @zeGolem  Just stumbled on this project… Would be cool to have build instructions for advanced users who don't want to rely on the prebuilt binaries, or want to test features before a new release
The best I can suggest is just git clone the repo, npm install, npm start that set this is how you can test new features before the release why you want to build an executable?

**Comment by zeGolem** [comment]
> > @zeGolem  Just stumbled on this project… Would be cool to have build instructions for advanced users who don't want to rely on the prebuilt binaries, or want to test features before a new release
> 
> The best I can suggest is just git clone the repo, npm install, npm start that set this is how you can test new features before the release why you want to build an executable?

That's the kind of things that I wanted in the repo, and it was added :) As for creating an executable, the main reason would be for someone wanting to package this app in say the AUR, or any distro's repository. Typically, it's frowned upon to rely on someone else's built executable for creating a package, so having easy instructions for creating one makes it more likely it'll get picked up by a maintainer :)

**Comment by Rikysonic** [comment]
> `android-irface.sh` allow you to delete cache (deleting the whole revanced/), but this deletes the keystore too. I hope to see an option to preserve the keystore while removing the real "cache". P.S. I'm talking about before running the revanced-builder instead during reinstallation.

I've opened a PR to add this 🙂

**Comment by Amby713** [comment]
It would be nice if you put the versions of the patches , somewhere, that way we'd know if we need to update or not

**Comment by misikovich** [comment]
ReVanced Music sleep timer patch

**Comment by reisxd** [comment]
> ReVanced Music sleep timer patch

Patches aren't something I can add. Suggest that in revanced/revanced-suggestions by creating a discussion.

**Comment by Amby713** [comment]
> > ReVanced Music sleep timer patch
> 
> Patches aren't something I can add. Suggest that in revanced/revanced-suggestions by creating a discussion.

No, not adding patches. What i meant was your script grabbing the version number and displaying it so that when I run it, I can see what tune version number is. This would give me a idea whether I need to continue to upgrade or quit because i already have the latest version

**Comment by Naash21** [comment]
Can i have a permanent repeat toggle for youtube music please i love to hear songs.

**Comment by reisxd** [comment]
> Can i have a permanent repeat toggle for youtube music please i love to hear songs.

That's litterally a core feature...? And also not the place to suggest patches.

**Comment by plasma3000** [comment]
Can you make the patched file to be created in the "export" folder?

**Comment by twistedmacaroni** [comment]
inbuilt video-downloader for reddit?

**Comment by Quahis** [comment]
revanced builder docker image

**Comment by shrihanDev** [comment]
> revanced builder docker image

It was actually there but now removed cuz no one was testing it and was kinda buggy.

**Comment by Broukmiken** [comment]
Hello,
an automatic build (latest patch, latest YT, all checked by default ) without the need to click "continue"
i mean an manual build and an automatic one.
Thanks so much for all the job you've done for us.

**Comment by reisxd** [comment]
> an automatic build (latest patch, latest YT, all checked by default ) without the need to click "continue"

Builder shouldn't be fully automatic.

**Comment by harshji888** [comment]
> Hi everyone, I want to implement new and awesome things to Builder so, I ask you, what do you want me to implement to Builder?
> 
> Also, soon there'll be an native Android app to build ReVanced even more easily (if i don't fail) ;)

Please tell how to make clone of YouTube revanced it's say it was a Google dependency app

**Comment by harshji888** [comment]
![Uploading Screenshot_2022-09-18-17-57-03-320_com.applisto.appcloneR.jpg…]()

**Comment by mmdhsiam** [comment]
Need to add youtube shorts and Autoplay. Also When a video plays for a few seconds in preview, after clicking it comes from the time it was played. I want it to play from the beginning after clicking.

**Comment by seaque** [comment]
Text search area for patches.

**Comment by reisxd** [comment]
> Text search area for patches.

Added ;)

The search box kinda looks ugly and well, I'm not a frontend developer so someone else can improve that (;

**Comment by reisxd** [comment]
> Install patched apk on multiple devices (root) without rebuild

Already exists. Plug the devices you have to your PC and start builder.

**Comment by JuliusDer** [comment]
Support for setting the options of selected patches like the ones below
[
   {
      "key":"granularity",
      "title":"Video speed granularity",
      "description":"The granularity of the video speeds. The higher the value, the more speeds will be available.",
      "required":true,
      "choices":null
   },
   {
      "key":"min",
      "title":"Minimum video speed",
      "description":"The minimum video speed.",
      "required":true,
      "choices":null
   },
   {
      "key":"max",
      "title":"Maximum video speed",
      "description":"The maximum video speed. Must be greater than the minimum video speed and smaller than 5.",
      "required":true,
      "choices":null
   }
]

**Comment by reisxd** [comment]
> Support for setting the options of selected patches like the ones below
> 
> ```
> [
>    {
>       "key":"granularity",
>       "title":"Video speed granularity",
>       "description":"The granularity of the video speeds. The higher the value, the more speeds will be available.",
>       "required":true,
>       "choices":null
>    },
>    {
>       "key":"min",
>       "title":"Minimum video speed",
>       "description":"The minimum video speed.",
>       "required":true,
>       "choices":null
>    },
>    {
>       "key":"max",
>       "title":"Maximum video speed",
>       "description":"The maximum video speed. Must be greater than the minimum video speed and smaller than 5.",
>       "required":true,
>       "choices":null
>    }
> ]
> ```

It's currently not implemented properly. So I also cannot implement it.

**Comment by Vretu-Dev** [comment]
Spotify patch ❤️

**Comment by reisxd** [comment]
> Spotify patch ❤️

Spotify is not on APKMirror because of receiving a DMCA by Spotify.

**Comment by Vretu-Dev** [comment]
> > Spotify patch ❤️
> 
> Spotify is not on APKMirror because of receiving a DMCA by Spotify.

What do you think about APKPure 
https://m.apkpure.com/pl/spotify-music-i/com.spotify.music

**Comment by reisxd** [comment]
> > > Spotify patch ❤️
> > 
> > Spotify is not on APKMirror because of receiving a DMCA by Spotify.
> 
> What do you think about APKPure 
> https://m.apkpure.com/pl/spotify-music-i/com.spotify.music

APKPure does not give an APK file, but a link to GPlay.

**Comment by Vretu-Dev** [comment]
@reisxd 
I found https://apkcombo.com/pl/spotify/com.spotify.music/download/apk

**Comment by Vretu-Dev** [comment]
Thats looks like legit
![image](https://user-images.githubusercontent.com/82831707/192150605-c54e5d8d-b9bb-46e5-9e8f-9ac2ded69be4.png)

**Comment by StandingPadAnimations** [comment]
Uploading custom icons for the custom branding patch

**Comment by Vretu-Dev** [comment]
Version on linux which will automatically download the apk when a new patch comes out, useful on servers

**Comment by reisxd** [comment]
> Version on linux which will automatically download the apk when a new patch comes out, useful on servers

https://github.com/reisxd/revanced-builder/issues/283#issuecomment-1250220339

**Comment by SimPilotAdamT** [comment]
What happened to the AMOLED black theme? Will it return?

**Comment by luxysiv** [comment]
Easy way to build on Android. Fork your repo. Create a codespace on fork. In terminal type `node .` Done

**Comment by TgZ39** [comment]
> What happened to the AMOLED black theme? Will it return?

More like how do I get rid of it.

**Comment by SimPilotAdamT** [comment]
> More like how do I get rid of it.

I just found out that it's labelled `Custom Theme`... Please don't remove it, that theme is probably one of the best features Vanced had to offer (and ReVanced has to offer). The stock Google dark theme is probably one of the shittest dark themes around.

**Comment by TinyVanced** [comment]
Can't you add an alternative site other than APKMirror to the site where you download the apk?
At least APKMirror is blocked by China, so I have to use a VPN
This is just a suggestion. If it seems impossible, just ignore it

**Comment by SimPilotAdamT** [comment]
> Can't you add an alternative site other than APKMirror to the site where you download the apk? At least APKMirror is blocked by China, so I have to use a VPN This is just a suggestion. If it seems impossible, just ignore it

APKMirror is probably the most reliable source... Maybe it's possible to get it to download from an actual Chinese mirror? Smth like how there's a version of the Arch User Repository hosted in China so the Chinese users can use the AUR just fine?

**Comment by TgZ39** [comment]
> > More like how do I get rid of it.
> 
> I just found out that it's labelled `Custom Theme`... Please don't remove it, that theme is probably one of the best features Vanced had to offer (and ReVanced has to offer). The stock Google dark theme is probably one of the shittest dark themes around.

Google dark theme is the best

**Comment by SimPilotAdamT** [comment]
> Google dark theme is the best

>:(

**Comment by amanrajOO7** [comment]
Can you implement patch options?

**Comment by reisxd** [comment]
> Can you implement patch options?

You can read the other comments.

**Comment by harshji888** [comment]
Features to get 1000 view by seeing 1 video 😂😂

**Comment by harshji888** [comment]
And 4000 watch hour time also

**Comment by JoaoHamerski** [comment]
Suggestion to add Facebook to your packages, there's already a "debullshified" version of it done on XDA.

More info here: https://forum.xda-developers.com/t/app-mod-facebook-lc-reborn-lcr.4420057/

This is a very old project that was descontinued and now some modder is trying to bring it back, he already did a great job, maybe you guys can do some help and bring it as a package here with more enhancements.

The official telegram an all info is in the link above.

**Comment by reisxd** [comment]
> 

I don't create patches. Also, Facebook is a React Native app, which can't be patched by the ReVanced Patcher.

**Comment by greenconda** [comment]
If app isn't installed on device, then download install it automatically before patching for root method

**Comment by anxiouslelimo** [comment]
> I would like to see the ability to choose older versions of Revanced Patcher, CLI and ReVanced Integrations, instead of downloading and using only new ones. New versions are not always stable.

Is this under consideration?

**Comment by BlasterOverlord** [comment]
Support for rooted devices.
I don't want to install a copy of the YouTube app with the patches applied.

Since I have root, I want it to apply the patches on the stock YouTube app already installed on my device.

**Comment by reisxd** [comment]
> Support for rooted devices.
> I don't want to install a copy of the YouTube app with the patches applied.
> 
> Since I have root, I want it to apply the patches on the stock YouTube app already installed on my device.

Already exists.

**Comment by BlasterOverlord** [comment]
> > Support for rooted devices.
> > I don't want to install a copy of the YouTube app with the patches applied.
> > Since I have root, I want it to apply the patches on the stock YouTube app already installed on my device.
> 
> Already exists.

Does that mean, the patches will work even when I update my YouTube app from play store?

**Comment by reisxd** [comment]
> > > Support for rooted devices.
> > > I don't want to install a copy of the YouTube app with the patches applied.
> > > Since I have root, I want it to apply the patches on the stock YouTube app already installed on my device.
> > 
> > 
> > Already exists.
> 
> Does that mean, the patches will work even when I update my YouTube app from play store?

Nope.

**Comment by Kiina** [comment]
> Builder shouldn't be fully automatic.

I get that but is there any reason why I have to click continue when it finished downloading the deps/apk? I feel like unless something went wrong it should just jump to the next point.

**Comment by git-Hmmm** [comment]
> I get that but is there any reason why I have to click continue when it finished downloading the deps/apk? I feel like unless something went wrong it should just jump to the next point.

Hi there, check this out #538, I hope reis will merge this one.

**Comment by Naash21** [comment]
Yes please

**Comment by singhdilveer1** [comment]
Probably **skip the platform checks** to make sure Revanced Builder binaries work on older Windows systems like Windows 7. It might be unstable but at least it'll work. 

**Showing users their JAVA_HOME location** whenever a version error occurs will also help. 

UI is already pretty cool. But since you asked. a **Grid UI option would be nice to have, it'd look a lot cleaner. **

**Comment by reisxd** [comment]
> Probably **skip the platform checks** to make sure Revanced Builder binaries work on older Windows systems like Windows 7. It might be unstable but at least it'll work. 

As you said, it would be unstable. Have you tried fully building ReVanced on a W7 VM/PC?

> **Showing users their JAVA_HOME location** whenever a version error occurs will also help. 

Builder tries to run the `java` command and if it can't run it, it'll throw an error. If the version is older than JDK 17, it'll also throw an error. The only way to fix this is to just change your PATH.

> UI is already pretty cool. But since you asked. a **Grid UI option would be nice to have, it'd look a lot cleaner. **

Grid UI? Sorry, I am not a frontend developer.

**Comment by singhdilveer1** [comment]
> As you said, it would be unstable. Have you tried fully building ReVanced on a W7 VM/PC?
> 
Actually yes, I have. It works pretty well, for now at least, it hasn't been too long since Windows 7 support dropped so I think it'll take a while for Node to completely stop working on W7. You could just add a warning that older Windows versions won't be supported and no help will be provided for the same.

> Builder tries to run the `java` command and if it can't run it, it'll throw an error. If the version is older than JDK 17, it'll also throw an error. The only way to fix this is to just change your PATH.
> 
I recently came across the [issue #541 ](https://github.com/reisxd/revanced-builder/issues/541). It seems more like a user error but still it'd help those who have multiple java versions conflicting with each other.

> Grid UI? Sorry, I am not a frontend developer.

Oh, well didn't realize that. It's fine, as I said UI is already pretty cool. 😉

**Comment by singhdilveer1** [comment]
> Well, I have to implement a function to run every single JDK/JRE installation and find the one that's compatible.
> 
Do you plan to? I know it's going to be a hassle.




> Bad news. We have to recompile NodeJS to disable that check from Node. [See this comment from `pkg` repo](https://github.com/vercel/pkg/issues/1156#issuecomment-835194591).
>
I think that would be a pain, recompiling every time you want to update NodeJS versions. I didn't really have any idea how this all works while writing the suggestion. Maybe add a line in the readme about the NODE_SKIP_PLATFORM_CHECK variable. It'll help all the newbies like me who have to search around for fixes. Because on Windows 7, the builder just crashes without any information, if run without the variable set.

**Comment by Vretu-Dev** [comment]
Maybe twitch

**Comment by LordBli** [comment]
Is there a possibility to remove the YouTube update from the Play Store ?

**Comment by git-Hmmm** [comment]
@LordBli 
![image](https://user-images.githubusercontent.com/102938585/204399955-b11e17de-de05-40a2-a240-e348a6bb7806.png)

**Comment by LordBli** [comment]
> @LordBli ![image](https://user-images.githubusercontent.com/102938585/204399955-b11e17de-de05-40a2-a240-e348a6bb7806.png)

@git-Hmmm 

I do not auto update apps but I can always see YouTube in the update section. Is there a way to hide it?
![Screenshot_2022-11-29-00-17-55-474_com android vending](https://user-images.githubusercontent.com/39700540/204400978-c84c9221-1eee-4dbb-9e0b-ed63c6a4b53e.png)

**Comment by git-Hmmm** [comment]
@LordBli Sadly no, there's no way I am aware of...

**Comment by reisxd** [comment]
@LordBli Use the detach module.

**Comment by LordBli** [comment]
> @LordBli Use the detach module.

@reisxd from magisk ?

**Comment by THB180** [comment]
Please add 
1. Preferred video speed
2. Preferred video quality
3. Downloading 1080/2k unlocked for 720p resolution mobile
4. Copy link with timestamp
5. Video sort by date oldest options.
This is a great challenge. Think you can handle it.

**Comment by rorosaurus** [comment]
> > Support for setting the options of selected patches like the ones below
> > ```
> > [
> >    {
> >       "key":"granularity",
> >       "title":"Video speed granularity",
> >       "description":"The granularity of the video speeds. The higher the value, the more speeds will be available.",
> >       "required":true,
> >       "choices":null
> >    },
> >    {
> >       "key":"min",
> >       "title":"Minimum video speed",
> >       "description":"The minimum video speed.",
> >       "required":true,
> >       "choices":null
> >    },
> >    {
> >       "key":"max",
> >       "title":"Maximum video speed",
> >       "description":"The maximum video speed. Must be greater than the minimum video speed and smaller than 5.",
> >       "required":true,
> >       "choices":null
> >    }
> > ]
> > ```
> 
> It's currently not implemented properly. So I also cannot implement it.

It looks like PatchOptions are working these days. :D Support for this would be really nice!

**Comment by git-Hmmm** [comment]
Yea, I guess a popup with checkboxes or some input text field would be really great idea.

**Comment by reisxd** [comment]
> > > Support for setting the options of selected patches like the ones below
> > > ```
> > > [
> > >    {
> > >       "key":"granularity",
> > >       "title":"Video speed granularity",
> > >       "description":"The granularity of the video speeds. The higher the value, the more speeds will be available.",
> > >       "required":true,
> > >       "choices":null
> > >    },
> > >    {
> > >       "key":"min",
> > >       "title":"Minimum video speed",
> > >       "description":"The minimum video speed.",
> > >       "required":true,
> > >       "choices":null
> > >    },
> > >    {
> > >       "key":"max",
> > >       "title":"Maximum video speed",
> > >       "description":"The maximum video speed. Must be greater than the minimum video speed and smaller than 5.",
> > >       "required":true,
> > >       "choices":null
> > >    }
> > > ]
> > > ```
> > 
> > It's currently not implemented properly. So I also cannot implement it.
> 
> It looks like PatchOptions are working these days. :D Support for this would be really nice!

Thing is, there are no PatchOption types on the JSON. So, Builder has no choice but assume every PatchOption requires a string or hardcoding it, which isn't a great solution. Still waiting for PatchOptions to have it's type in the JSON.

**Comment by rorosaurus** [comment]
> > > > Support for setting the options of selected patches like the ones below
> > > 
> > > 
> > > It's currently not implemented properly. So I also cannot implement it.
> > 
> > 
> > It looks like PatchOptions are working these days. :D Support for this would be really nice!
> 
> Thing is, there are no PatchOption types on the JSON. So, Builder has no choice but assume every PatchOption requires a string or hardcoding it, which isn't a great solution. Still waiting for PatchOptions to have it's type in the JSON.

I suppose another option could be to create a dropdown for the type, forcing the end user to own that type responsibility for any options they choose to provide, based on the option description/title. But I understand why that doesn't feel super elegant.

**Comment by chunkyjackson** [comment]
If possible, make it possible to upgrade non root without reinstalling.

**Comment by shrihanDev** [comment]
> If possible, make it possible to upgrade non root without reinstalling.

It's not possible, unfortunately. Non-root ReVanced is just a regular app, and of course needs a reinstall for updating.

**Comment by Niloy-Sarker** [comment]
split apks support for youtube?

**Comment by git-Hmmm** [comment]
PatchOptions for all supported apps:
a tune icon at every patch right corner that opens this popup
![image](https://user-images.githubusercontent.com/102938585/208704405-c17c6ed4-1aff-428f-8004-25f9d3c7ad31.png)
maybe something like this can be done.

currently it targets advanced users only, but when there are PatchOption types on the JSON available this can be redesigned.

**Comment by reisxd** [comment]
> split apks support for youtube?

Not our job to do. Ask that to ReVanced.

**Comment by araafroyall** [comment]
Hi i want a feature to download youtube videos from app directly in internal storage so can be viewed by other apps

**Comment by mmdhsiam** [comment]
If the possible a create app where we can patching and downloa easily? like Revanced manager. That will be great for us.

**Comment by DrimTim2020** [comment]
Where did the dark theme go?
There is only Black Amoled.

**Comment by nik447n** [comment]
patches for Truecaller and spotify music

**Comment by Fisiu** [comment]
What about import/export selected patches? After recreating container, we start with blank patches' list.

**Comment by DrimTim2020** [comment]
> 

Already after the first application, all selected patches are saved.

**Comment by DrimTim2020** [comment]
> What about import/export selected patches? After recreating container, we start with blank patches' list.

After the first application, all selected patches are saved.

**Comment by Fisiu** [comment]
> > What about import/export selected patches? After recreating container, we start with blank patches' list.
> 
> After the first application, all selected patches are saved.

That's true but only for the same container. If you ie. update an image and create a new container, all the previous selection is lost.

**Comment by DrimTim2020** [comment]
> That's true but only for the same container. If you ie. update an image and create a new container, all the previous selection is lost.

It's most likely your problem.
For example, it doesn’t matter what I do, everything is saved to a file ( settings.json ) this is the same save file.
And any of the programs I choose for the first time, I choose patches; for the second time, they are already automatically substituted.

**Comment by Fisiu** [comment]
It seems that settings file is created under `/app/revanced-builder/settings.json` but volume (in provided `docker-compose.yml` is mounted under `/app/revanced-builder/revanced/`. So it's not possible to retain settings when re-creating container.

Shouldn't settings file be included in a volume?

**Comment by DrimTim2020** [comment]
I would like to see what is really needed in the program.
This is so that the program can patch the latest versions of Youtube _ YouTube Music _ Tik-Tok and so on that are installed on the phone, and not the old versions.
And it also applies to Revanced menager, for example, I have 18.04.35 installed and it offers me 18.03.36.

**Comment by AniketDeshmane** [comment]
Hi Devs,
Thank for what you are doing but could you please add a setting export feature in vanced


Thanks

**Comment by memtos965** [comment]
There is such a wonderful translator from the Yandex browser that translates videos from YouTube well, maybe someone can integrate into revanced. There is a developer who has created code that integrates translation functions into other browsers without reference to Yandex. https://github.com/ilyhalight/voice-over-translation

**Comment by mookeyj79** [comment]
So, unfortunately the version of yt compatible with the patches has aged off the first page of apkmirror. An option to look releases on other pages would be nice. Also, I tried the "upload apk" option, but seems to hang at the "Upload successful, please wait" screen.

**Comment by ShaptakNaskar** [comment]
The implementation of an option that enables users to specify their preferred application name instead of the default YouTube ReVanced Name. Additionally a feature that allows the use of custom icons instead of the default YouTube ReVanced icon

**Comment by Joly0** [comment]
Dont know, if this has already been asked for, but i would like to have a feature for "Templating" or saving "user-profiles".

So User A patches App 1 in a certain way User B doesnt want it to be patched. Would be great to have some "save profile" and "load profile" functionality, so i can load my previous selection or a saved selection for a certain app, which other users could load aswell, but could also save others for theirselfs

**Comment by nik447n** [comment]
I think this is available in revanced manager (Android app )

**Comment by Zoey2936** [comment]
An option to set the listing port via start argument (like --no-open) would be nice.

**Comment by Zoey2936** [comment]
an option to see older apk versions would be nice, since the latest working Instagram version is so old that it doesn't show under the versions to download
